src/game/client/components/spectator.cpp

```diff
@@ -171,10 +171,12 @@ bool CSpectator::OnInput(const IInput::CEvent &Event)
 		return true;
 	}
 
-	if(g_Config.m_ClSpectatorMouseclicks)
+	if(g_Config.m_ClSpectatorMouseclicks && m_pClient->m_Snap.m_SpecInfo.m_Active && !IsActive() && !GameClient()->m_MultiViewActivated &&
+		!Ui()->IsPopupOpen() && !m_pClient->m_GameConsole.IsActive() && !m_pClient->m_Menus.IsActive())
 	{
-		if(m_pClient->m_Snap.m_SpecInfo.m_Active && !IsActive() && !GameClient()->m_MultiViewActivated &&
-			!Ui()->IsPopupOpen() && !m_pClient->m_GameConsole.IsActive() && !m_pClient->m_Menus.IsActive())
+		// 1 second delay in pvp modes to prevent accidental switching when killed
+		if(!GameClient()->m_GameInfo.m_Pvp || (m_pClient->m_Snap.m_pLocalCharacter ||
```

Comment:
```suggestion
		if(!GameClient()->m_GameInfo.m_SpecAfterKill || (m_pClient->m_Snap.m_pLocalCharacter ||
```

If we want to do this via gameflags: Gameflags should generally not be re-used. Please create a new one that specifies what the flag is supposed to represent. There are game modes where you don't get sent to spectators after being killed and I can imagine race modes where you are.

Reply:
What's the best way to do it? Zcatch mode isn't updated, so they won't add a new flag. Would something like this work?
`SpecAfterDeath = str_find_nocase(pGameType, "catch") || str_find_nocase(pGameType, "bomb");`

Reply:
Under `Info.m_DDRaceTeam = false;`, you can initialize `Info.m_SpecAfterDeath = !Vanilla && !Race;`, then parse it normally by introducing a new info version and a new gameflag.

Reply:
Could you change that to use `!Vanilla && !Race` instead of trying to parse the game type?

Reply:
Done, but I'm not sure if adding a new info version with a new game flag is suitable for a small fix like this